hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java

```diff
@@ -960,428 +909,1334 @@ void getTokenInfo(final TokenTypeEnum tokenType) throws Exception {
-        final var result = contractCallService.processCall(serviceParameters);
-        assertThat(result).isEqualTo(expectedResult);
-    }
+        final var contractId = testWeb3jService.getEntityId();
+        persistAssociation(tokenEntity, contractId);
 
-    @Test
-    void nftInfoForInvalidSerialNo() {
-        // Given
-        final var functionHash = functionEncodeDecoder.functionHashFor(
-                "getInformationForNonFungibleToken", PRECOMPILE_TEST_CONTRACT_ABI_PATH, NFT_ADDRESS, 4L);
+        domainBuilder
+                .tokenAllowance()
+                .customize(ta -> ta.tokenId(tokenEntity.getId())
+                        .spender(testWeb3jService.getEntityId())
+                        .amount(10L)
+                        .owner(spender.getId()))
+                .persist();
 
         // When
-        final var serviceParameters = serviceParametersForExecution(
-                functionHash, PRECOMPILE_TEST_CONTRACT_ADDRESS, ETH_CALL, 0L, BlockType.LATEST);
+        final var functionCall = contract.call_transferFrom(
+                getAddressFromEntity(tokenEntity),
+                getAliasFromEntity(spender),
+                getAliasFromEntity(recipient),
+                BigInteger.ONE);
 
         // Then
-        assertThatThrownBy(() -> contractCallService.processCall(serviceParameters))
-                .isInstanceOf(MirrorEvmTransactionException.class);
+        verifyEthCallAndEstimateGas(functionCall, contract, ZERO_VALUE);
     }
 
-    @Test
-    void tokenInfoForNonTokenAccount() {
+    @ParameterizedTest
+    @CsvSource("1, 0")
+    void approve(final BigInteger allowance) throws Exception {
         // Given
-        final var functionHash = functionEncodeDecoder.functionHashFor(
-                "getInformationForFungibleToken", PRECOMPILE_TEST_CONTRACT_ABI_PATH, SENDER_ADDRESS);
+        final var spender = persistAccountEntity();
+        final var tokenEntity = persistFungibleToken();
+
+        persistAssociation(tokenEntity, spender);
+
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+
+        final var contractId = testWeb3jService.getEntityId();
+        persistAssociation(tokenEntity, contractId);
 
         // When
-        final var serviceParameters = serviceParametersForExecution(
-                functionHash, PRECOMPILE_TEST_CONTRACT_ADDRESS, ETH_CALL, 0L, BlockType.LATEST);
+        final var functionCall = contract.call_approveExternal(
+                getAddressFromEntity(tokenEntity), getAddressFromEntity(spender), allowance);
 
         // Then
-        assertThatThrownBy(() -> contractCallService.processCall(serviceParameters))
-                .isInstanceOf(MirrorEvmTransactionException.class);
+        verifyEthCallAndEstimateGas(functionCall, contract, ZERO_VALUE);
     }
 
-    @Test
-    void notExistingPrecompileCallFails() {
+    @ParameterizedTest
+    @CsvSource("true, false")
```

Comment:
Same comment elsewhere

```suggestion
    @ValueSource(booleans = {true, false})
```

Reply:
Fixed